src/searching/saddleback_search.rs

```diff
@@ -1,85 +1,147 @@
-// Saddleback search is a technique used to find an element in a sorted 2D matrix in O(m + n) time,
-// where m is the number of rows, and n is the number of columns. It works by starting from the
-// top-right corner of the matrix and moving left or down based on the comparison of the current
-// element with the target element.
 use std::cmp::Ordering;
 
-pub fn saddleback_search(matrix: &[Vec<i32>], element: i32) -> (usize, usize) {
-    // Initialize left and right indices
+/// Performs Saddleback search on a sorted 2D matrix.
+///
+/// The Saddleback search algorithm finds the position of a target element in a matrix where
+/// each row and each column is sorted in ascending order. The search starts from the top-right
+/// corner of the matrix and moves left or down based on comparisons with the target element.
+///
+/// # Arguments
+///
+/// * `matrix` - A 2D vector representing the sorted matrix.
+/// * `element` - The target element to search for.
+///
+/// # Returns
+///
+/// Returns `Some((row, column))` where both indices are 0-based. If the element is not found, returns `None`.
+pub fn saddleback_search(matrix: &[Vec<isize>], element: isize) -> Option<(usize, usize)> {
```

Comment:
What if the input is not a proper matrix? (i.e. not a _rectangle_).

Reply:
In the case of a proper matrix, we can return an error indicating that the matrix is not rectangular.